format_spec.md

```diff
@@ -0,0 +1,399 @@
+
+The following fields are changed fields for dependencies. Dependencies MUST contain each of these fields. These fields are the minimum requirement for a dependency.
+
+##### `id`
+
+The ID of the required dependency. If this project is a Framework, it MUST be one of the values listed in **format_values.md** for Framework IDs. Hosts serving Framework metadata MUST use these standardized IDs.
+
+##### `allowed` (optional)
+
+Same as `version.allowed` except for this dependency.
+
+##### `required`
+
+This field is a boolean that indicates whether a dependency is required in order for the Project requiring it to function properly. **MAY be a condition.**
+
+##### `version`
+
+This field MUST be either an Array or String. If this field is an array, it MUST contain a list of compatible versions. If this field is a String, it MUST contain a Semantic Versioning comparison String. An example of a comparison string is `>= 25.0.219`.
+
+##### `src` (optional)
+
+If not all information is listed inside a dependency object and it's metadata is not hosted on the same host which is serving metadata for the parent project, the `src` field may be used. This field MUST contain a URL. This URL MUST serve MODIP Format-compliant metadata about the required dependency. It MUST include the `versions` field.
+
+<br>
+
+The required Minecraft Version also MUST be stored as a dependency. It's `id` value MUST be set to `minecraft`. It also MUST not contain the `required` field, as Minecraft is not an optional component.
+
+A dependencies implementation guide for launchers is available in **format_implementing.md**.
+
+---
+
+### `files`
+
+This field contains the downloadable files of a Version. It MUST be an array of File Objects.
+File Objects contain information on a downloadable file. Fields are as follows:
+
+##### `name`
+
+There is no restriction on what this value can be. However, it is RECOMMEND that this be the name of the file. Launchers and other automation tools MAY ignore the value. This value CAN be shown to users.
+
+##### `sha256`
+
+An SHA256 checksum of the file. This may be used by launchers in order to verify download integrity.
+
+##### `rel`
+
+The relation of this file. Possible values for this field are available in **format_values.md**. The values of this field CAN be present in multiple files. For example, a version can have a file for both the Forge and Fabric modloaders, and both of those files can have the `rel` field set to `primary`.
+
+##### `dependencies` (optional)
+
+This field MUST conform to the same specification as the Version's dependencies. BOTH a File and Version can have dependencies.
+
+It is RECOMMENDED to define Framework/Modloader dependencies as File Dependencies and NOT Version Dependencies. When downloading a file, launchers SHOULD pay attention to both the Version and FIle dependencies.
```

Comment:
Why is that recommended, if they're the same for all files?

Reply:
It's recommended for consistency with mods supporting multiple modloaders, but it's probably unnecessary and I may remove it.